File: app/src/main/java/com/code2concept/blurbehindalertdialog/MainActivity.java
package com.code2concept.blurbehindalertdialog;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.DialogInterface;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import com.code2concept.blurbehindalertdialog.utils.AppUtils;
import com.code2concept.blurbehindalertdialog.views.BlurView;

public class MainActivity extends Activity {

    private AlertDialog alert;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        findViewById(R.id.blurButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
                builder.setTitle("Code2Concept - Blur");
                builder.setMessage("This is Blur Demo.Click Ok to exit");
                builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.cancel();
                    }
                });


                alert = builder.create();

               new BlurAsyncTask().execute();

            }
        });
    }

    class BlurAsyncTask extends AsyncTask<Void, Integer, Bitmap> {

        private  final String TAG = BlurAsyncTask.class.getName();

        protected Bitmap doInBackground(Void...arg0) {

            Bitmap map  = AppUtils.takeScreenShot(MainActivity.this);
            Bitmap fast = new BlurView().fastBlur(map, 10);

            return fast;
        }


        protected void onPostExecute(Bitmap result) {
            if (result != null){
                final Drawable draw=new BitmapDrawable(getResources(),result);
                Window window = alert.getWindow();
                window.setBackgroundDrawable(draw);
                window.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
                window.setGravity(Gravity.CENTER);
                alert.show();

                TextView textView = (TextView) alert.findViewById(android.R.id.title);
                if (textView != null)
                    textView.setGravity(Gravity.CENTER);
            }

        }
    }
}

File: app/src/main/java/com/code2concept/blurbehindalertdialog/MainActivity.kt
package com.code2concept.blurbehindalertdialog

import android.app.Activity
import android.app.AlertDialog
import android.content.Context
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.os.AsyncTask
import android.os.Bundle
import android.view.Gravity
import android.view.Window
import android.view.WindowManager
import android.widget.Button
import android.widget.TextView

import com.code2concept.blurbehindalertdialog.utils.AppUtils
import com.code2concept.blurbehindalertdialog.views.BlurView

class MainActivity : Activity() {

    private var alert: AlertDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        findViewById<Button>(R.id.blurButton).setOnClickListener {
            val builder = AlertDialog.Builder(this@MainActivity)
            builder.setTitle("coderconsole.com")
            builder.setMessage("BlurBackground of dialog using renderscript api. \nClick Ok to Exit ")
            builder.setPositiveButton(getString(android.R.string.ok)) { dialog, which -> dialog.cancel() }

            alert = builder.create()

            alert?.findViewById<TextView>(android.R.id.title)?.gravity = Gravity.CENTER


            Thread(Runnable {
                AppUtils.screenShot(this){bitmap ->
                    val blurBitmap: Bitmap? = BlurView(application).blurBackground(bitmap, 10)
                    runOnUiThread {
                        setAlertDialogBackground(blurBitmap)
                    }
                }
            }).start()


        }
    }
    private fun setAlertDialogBackground(result: Bitmap?) {
        val draw = BitmapDrawable(resources, result)
        val window = alert?.window
        window?.setBackgroundDrawable(draw)
        window?.setLayout(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT)
        window?.setGravity(Gravity.CENTER)
        alert?.show()

        val textView = alert?.findViewById<TextView>(android.R.id.title)
        textView?.gravity = Gravity.CENTER
    }

}
